components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/CommentsManagerImpl.java

```diff
@@ -17,10 +17,274 @@
+import org.wso2.carbon.device.application.mgt.core.dao.CommentDAO;
+import org.wso2.carbon.device.application.mgt.core.dao.common.ApplicationManagementDAOFactory;
+import org.wso2.carbon.device.application.mgt.core.dao.common.Util;
+import org.wso2.carbon.device.application.mgt.core.exception.ApplicationManagementDAOException;
+import org.wso2.carbon.device.application.mgt.core.exception.NotFoundException;
+import org.wso2.carbon.device.application.mgt.core.util.ConnectionManagerUtil;
+
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
 
 /**
  * This class is the default implementation for the Managing the comments.
  */
+@SuppressWarnings( "deprecation" )
 public class CommentsManagerImpl implements CommentsManager {
-}
+
+    private static final Log log = LogFactory.getLog(CommentsManagerImpl.class);
+    private CommentDAO commentDAO;
+
+    public CommentsManagerImpl() {
+        initDataAccessObjects();
+    }
+
+    private void initDataAccessObjects() {
+        this.commentDAO= ApplicationManagementDAOFactory.getCommentDAO();
+    }
+
+    @Override
+    public Comment addComment(Comment comment,String uuid,int tenantId) throws CommentManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Request for comment is received for uuid" +uuid);
+        }
+        comment.setCreatedAt(Timestamp.from(Instant.now()));
+
+        try {
+            ConnectionManagerUtil.beginDBTransaction();
+            ApplicationManagementDAOFactory.getCommentDAO().addComment(tenantId,comment,
+                    comment.getCreatedBy(),comment.getParent(),uuid);
+            ConnectionManagerUtil.commitDBTransaction();
+            return comment;
+        } catch (Exception e) {
+            log.error("Exception occurs.", e);
+            ConnectionManagerUtil.rollbackDBTransaction();
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return comment;
+    }
+
+    /**
+     * To validate the pre-request of the comment
+     *
+     * @param apAppCommentId ID of the comment.
+     * @param comment comment needed to be validate.
+     * @return Application related with the UUID.
+     *
+     *
+     */
+
+    public Boolean validateComment(int apAppCommentId,String comment) throws CommentManagementException{
+
+        if (apAppCommentId <= 0) {
+            throw new CommentManagementException("Comment ID is null or negative. Comment id is a required parameter to get the " +
+                    "relevant comment.");
+        }
+
+        if(comment==null){
+            throw new NullPointerException("No comment is entered.");
+        }
+
+        try {
+            throw new NotFoundException("Comment with comment id "+apAppCommentId+" does not exit");
+        } catch (NotFoundException e) {
+            log.error("Not Found Exception occurs.", e);
+        }
```

Comment:
@NishadiPrasangini what is the purpose of throwing an exception and catching the same exception? Can you please explain how the comment validity is checked here?